### PR TITLE
Add test for financials negative case

### PR DESCRIPTION
**Pull request by msahebsara**

### CODE REVIEW

File: test_financials.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import unittest
+from unittest.mock import Mock
+from unittest.mock import patch
+import yfinance as yf
+
+class FinancialsTestCase(unittest.TestCase):
+    def setUp(self):
+        self.msft = yf.Ticker("MSFT")
```
Comment (HaochenGou) [inline test_financials.py:8]: seems not mock the data

Comment (msahebsara) [reply test_financials.py:8]: See the next line, line 9

File: test_financials.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        self.utils_mock = patch('yfinance.base.utils.get_json', autospec=True).start()
+        self.addCleanup(patch.stopall)
+
+    def tearDown(self):
+        pass
+
+    def test_get_financials_invalid_data_returns_empty_dataframe(self):
+        self.assertTrue(self.msft.financials.empty)
```
Comment (HaochenGou) [inline test_financials.py]: yf.Ticker("MSFT") get data from web and test if it empty, self.utils_mock seems not be used,

File: test_financials.py (hunk 1 above)
Comment (HaochenGou) [reply test_financials.py:8]: the line 8 yf.Ticker("MSFT") get data from web and in test function test if it empty, self.utils_mock seems not be used,

Comment (msahebsara) [reply test_financials.py:8]: The get_json call in utils is being mocked, so when get_financials() is called, it will indeed return empty.
Please review this to see how patch works: https://docs.python.org/3/library/unittest.mock.html

**Review by HaochenGou — APPROVED** [review]

File: test_financials.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        self.addCleanup(patch.stopall)
+
+    def tearDown(self):
+        pass
+
+    def test_get_financials_invalid_data_returns_empty(self):
+        try:
+            financials = self.msft.financials 
+            self.assertTrue(financials.empty)
+        except:
```
Comment (kpatenio) [inline test_financials.py:19]: I'm not quite sure if I understand why we need an except statement here.

Couldn't we just use lines 17-18 without a try-except? I ran your test without it and it seems to run fine (no failures)

Comment (kpatenio) [reply test_financials.py:19]: oops just saw that this was merged lol

Comment (msahebsara) [reply test_financials.py:19]: There should be a comment on line 20, but the reason for the try/except is so to ensure an exception is NOT raised because we want graceful behavior of the application. We should not be throwing an exception even with invalid data, but simply returning an empty property for financials.

Comment (kpatenio) [reply test_financials.py:19]: Hmmm I see, so it's more so to ensure the test runs smoothly then and not throw an exception since that's not its intended behaviour. I think that's more clear now, thanks 👍
